yt_dlp/postprocessor/sponsorblock.py

```diff
@@ -0,0 +1,391 @@
+from __future__ import unicode_literals
+
+import io
+import json
+import os
+import re
+from collections import OrderedDict
+from hashlib import sha256
+
+from . import FFmpegPostProcessor
+from ..compat import compat_urllib_parse_urlencode, compat_HTTPError
+from ..utils import (
+    encodeFilename,
+    float_or_none,
+    PostProcessingError,
+    prepend_extension,
+    sanitized_Request
+)
+
+CATEGORY_DISPLAY_NAMES = {
+    'sponsor': 'Sponsor',
+    'intro': 'Intro',
+    'outro': 'Outro',
+    'interaction': 'Interaction Reminder',
+    'selfpromo': 'Self-Promotion',
+    'music_offtopic': 'Non-Music Section'
+}
+
+
+class SponsorBlockPP(FFmpegPostProcessor):
+    ALL_CATEGORIES = 'sponsor', 'intro', 'outro', 'interaction', 'selfpromo', 'music_offtopic'
+    # See https://ffmpeg.org/general.html#Subtitle-Formats.
+    SUPPORTED_SUBS = 'srt', 'ass', 'vtt'
+
+    def __init__(self, downloader, cut=False, add_keyframes=False, force=False, plaintext=False,
+                 api='https://sponsor.ajay.app', categories=ALL_CATEGORIES):
```

Comment:
You'll want to also make sure people can choose to keep the modification time of the original file (i.e: `-no-id` functionality that exists in Sponskrub)

Reply:
Modification time is handled by [`real_run_ffmpeg`](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/postprocessor/ffmpeg.py#L244), which I don't touch at all.

[`no-id` from SponSkrub](https://github.com/faissaloo/SponSkrub/blob/master/src/sponskrub/sponskrub.d#L153) is mapped unto [`plaintext`](https://github.com/yt-dlp/yt-dlp/pull/360/files/e5f8f4afa79d37571f6ce76b3b14c135891317cc#diff-1051e7d43a7be354ecfc1ddf7411d6f4c14382839df59df5573b3c1d76e67490R61). Probably I should rename it into `reveal_video_id` to make it clearer.